crates/biome_js_analyze/src/analyzers/nursery/no_namespace_import.rs

```diff
@@ -0,0 +1,70 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::JsImportNamespaceClause;
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow the use of namespace imports.
+    ///
+    /// Namespace imports might impact the efficiency of tree shaking, a process that removes unused code from bundles.
```

Comment:
Just wondering: have you some example where namespace imports are mot optimally handled?

Reply:
The `mod` in index.js below will be not handled optimally because it's literally used.

```js
// mod.js
export const foo = "foo";
export const bar = "bar";
```

```index.js
// index.js
import * as mod from "./mod";
console.log(mod); // The entire module is included in the bundle and is not optimized
```

Ref (Japanese article): https://qiita.com/uhyo/items/842e51e0d8cc46856d04

Reply:
You can find information here too https://marvinh.dev/blog/speeding-up-javascript-ecosystem-part-7/

Reply:
@unvalley 

In this case, it could be valuable to provide a rule that forbids the use of a namespace as a value.

Reply:
@ematipico I have scratched the surface of the article. It doesn't seem to talk about namespace imports.

Reply:
@Conaclos 
I agree that the impact of restricting namespace imports on tree shaking is limited. But, I still find rules against namespace imports beneficial. Using named imports, which only allow necessary values, can improve code maintainability (though it might be worth adjusting the lint rule's description).

Reply:
> Using named imports, which only allow necessary values, can improve code maintainability

I am a bit in disagreement here. I often find the use of namespace import more helpful because they clearly define the source of a value.

In a future version we could adjust the behavior; if most of the values of the namespace are used, then we could disable the rule.